server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/HistogramAggregatorFactory.java

```diff
@@ -61,24 +60,40 @@ public long minDocCount() {
     }
 
     @Override
-    protected Aggregator doCreateInternal(ValuesSource.Numeric valuesSource, Aggregator parent, boolean collectsFromSingleBucket,
+    protected Aggregator doCreateInternal(ValuesSource valuesSource, Aggregator parent, boolean collectsFromSingleBucket,
             List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) throws IOException {
         if (collectsFromSingleBucket == false) {
             return asMultiBucketAggregator(this, context, parent);
         }
-        return createAggregator(valuesSource, parent, pipelineAggregators, metaData);
+        if (valuesSource instanceof ValuesSource.Numeric) {
+            return createAggregator((ValuesSource.Numeric) valuesSource, parent, pipelineAggregators, metaData);
+        }
+        else if (valuesSource instanceof ValuesSource.Bytes) {
```

Comment:
Note to our future-selves: we should make sure this doesn't "bleed" over to other types that also use bytes... I think Strings will show up as `ValuesSource.Bytes` (from `ValueType.String` -> `ValuesSourceType.BYTES` -> `ValuesSource.Bytes`)

Unless that's handled elsewhere and we can ignore :)

Reply:
Oh, actually, we should have a more specific `ValuesSource` here, now that I've added a range specific choice.  Nice catch.